docs/smbios.md

```diff
@@ -0,0 +1,53 @@
+## Introduction
+
+The System Management BIOS (SMBIOS) specification defines data structures (and access methods) that can be used to read management information produced by the BIOS of a computer.
+
+This allows us to gather hardware information about the running system and use that as part of our labels.
+
+## How does elemental uses SMBIOS data?
+
+The registration client tries to gather SMBIOS data by running `dmidecode` during the initial registration of the node and that data is
+sent to the registration controller to use on interpolating different fields in the inventory that we create for that node.
+
+Currently, we support interpolating that data into the `#!yaml machineName` and the `#!yaml machineInventoryLabels` of a [machineRegistration](machineregistration-reference.md)
+
+The interpolation format is as follows:
+
+`{$KEY/VALUE}` and `${KEY/SUBKEY/VALUE}`
+
+This can be mixed with normal strings so `my-prefix-${KEY/VALUE}` would result into the resolved values with `my-prefix-` prefixed
+
+
+For example, having the following SMBIOS data:
+
+```
+System Information
+	Manufacturer: My manufacturer
+	Product Name: Awesome PC
+	Version: Not Specified
+	Serial Number: THX1138
+	Family: Toretto
+```
+
+And setting the `#!yaml machineName` to `serial-${System Information/Serial Number}` would result in the final value of `serial-THX1138`
+
+This is useful to generate automatic names for machines based on their hardware values, for example using the UUID or the Product name.
+Our default `#!yaml machineName` when the registration values are empty is `"m-${System Information/Manufacturer}-${System Information/Product Name}-${System Information/UUID}"`
```

Comment:
ok, I shoud remember to change here when I will be ready for https://github.com/rancher/elemental-operator/pull/145 (this is a note to self)